File: SIT-Test_Solution/20-12/Python3/Baekjoon15624_Fibonacci7.py
# Baekjoon15624_Fibonacci7.py
# https://www.acmicpc.net/problem/15624
import sys; input = sys.stdin.readline

MOD = 1_000_000_007

n = int( input() )

dp = [ 0, 1 ]
for i in range( 2, n + 1 ):
	dp.append( (dp[i - 1] + dp[i - 2]) % MOD )

print( dp[n] )

File: SIT-Test_Solution/20-12/C/Baekjoon15624_Fibonacci7.c
// Baekjoon15624_Fibonacci7.c
// https://www.acmicpc.net/problem/15624
#include <stdio.h>

#define MOD 1000000007

int main(void) {
	int n;
	scanf("%d", &n);

	int dp[1000001] = { 0, 1, 1, };
	for ( int i = 3; i <= n; i++ )
		dp[i] = (dp[i - 1] + dp[i - 2]) % MOD;
	printf( "%d", dp[n] );

	return 0;
}
